src/api.v2/controllers/sampleFileController.js

```diff
@@ -0,0 +1,55 @@
+const sqlClient = require('../../sql/sqlClient');
+
+const Sample = require('../model/Sample');
+const SampleFile = require('../model/SampleFile');
+
+const { getSampleFileUploadUrl } = require('../helpers/s3/signedUrl');
+const { OK } = require('../../utils/responses');
+const getLogger = require('../../utils/getLogger');
+
+const logger = getLogger('[SampleFileController] - ');
+
+const createFile = async (req, res) => {
+  const {
+    params: { experimentId, sampleId, sampleFileType },
+    body: { sampleFileId, size, metadata = {} },
+  } = req;
+  logger.log(`Creating sample file for experiment ${experimentId}, sample ${sampleId}, sampleFileType ${sampleFileType}`);
+
+  const newSampleFile = {
+    id: sampleFileId,
+    sample_file_type: sampleFileType,
+    size,
+    s3_path: sampleFileId,
+    upload_status: 'uploading',
+  };
+
+  let signedUrl;
+
+  await sqlClient.get().transaction(async (trx) => {
```

Comment:
Will transactions roll back automatically if it fails? or do we have to rollback manually?

Reply:
Ok, so we don't have to wrap this in a try/catch and rollback manually right